src/values/color.rs

```diff
@@ -3134,6 +3152,105 @@ where
   current.into()
 }
 
+fn calculate_contrast_color(base_luminance: f32, mut candidates: Vec<CssColor>) -> CssColor {
+  if candidates.is_empty() {
+    candidates.push(CssColor::RGBA(RGBA::new(0, 0, 0, 1.0)));
+    candidates.push(CssColor::RGBA(RGBA::new(255, 255, 255, 1.0)));
+  }
+
+  fn wcag2_contrast_ratio(l1: f32, l2: f32) -> f32 {
+    // https://www.w3.org/TR/WCAG21/#dfn-contrast-ratio
+    let l1 = l1 + 0.05;
+    let l2 = l2 + 0.05;
+
+    if l1 > l2 {
+      l1 / l2
+    } else {
+      l2 / l1
+    }
+  }
+
+  let mut best_contrast = 0.0;
+  let mut best_contrast_index = 0;
+
+  for (i, candidate) in candidates.iter().enumerate() {
+    let candidate_luminance = candidate.relative_luminance().unwrap();
+    let contrast = wcag2_contrast_ratio(base_luminance, candidate_luminance);
+
+    if contrast > best_contrast {
+      best_contrast = contrast;
+      best_contrast_index = i;
+    }
+  }
+
+  candidates[best_contrast_index].clone()
+}
+
+fn parse_contrast_color<'i, 't>(input: &mut Parser<'i, 't>) -> Result<CssColor, ParseError<'i, ParserError<'i>>> {
+  let base_color = CssColor::parse(input)?;
+
+  let base_luminance = match base_color.relative_luminance() {
+    Some(value) => value,
+    None => {
+      return Err(input.new_custom_error(ParserError::InvalidValue));
+    }
+  };
+
+  let location = input.current_source_location();
+
+  match input.expect_ident() {
+    Ok(value) => {
+      match_ignore_ascii_case! { value,
+        // https://drafts.csswg.org/css-color-5/#contrast-color
+        "max" => {
+          return Ok(calculate_contrast_color(base_luminance, Vec::new()));
+        },
+
+        // https://github.com/w3c/csswg-drafts/issues/7937
+        #[cfg(feature = "level6")]
+        "tbd-bg" | "tbd-fg" => {
+          input.expect_ident_matching("wcag2")?;
```

Comment:
Looks like this is optional in the spec.

> If no [color candidates](https://drafts.csswg.org/css-color-6/#contrast-color-candidates) have been provided, [<target-contrast>](https://drafts.csswg.org/css-color-6/#typedef-target-contrast) may be omitted, in which case a UA-chosen algorithm is used.

> If the [target contrast level](https://drafts.csswg.org/css-color-6/#target-contrast-level) is omitted, the color candidate with the greatest contrast is returned. Otherwise, the returned color is the first color candidate that meets or exceeds that level, defaulting to [white](https://drafts.csswg.org/css-color-4/#valdef-color-white) or [black](https://drafts.csswg.org/css-color-4/#valdef-color-black) if none qualify.

Also looks like it can be a function:

> Arguments to a [<target-contrast>](https://drafts.csswg.org/css-color-6/#typedef-target-contrast) [functional notation](https://drafts.csswg.org/css-values-4/#functional-notation) indicate the target contrast level.

> [<wcag2>](https://drafts.csswg.org/css-color-6/#typedef-wcag2) =  wcag2 [|](https://drafts.csswg.org/css-values-4/#comb-one) wcag2([[<number>](https://drafts.csswg.org/css-values-4/#number-value) [|](https://drafts.csswg.org/css-values-4/#comb-one) [ aa [|](https://drafts.csswg.org/css-values-4/#comb-one) aaa ] [&&](https://drafts.csswg.org/css-values-4/#comb-all) large[?](https://drafts.csswg.org/css-values-4/#mult-opt) ])